packages/shared/src/permissions/permissionsClass.ts

```diff
@@ -529,11 +529,19 @@ export class Permissions {
     );
   };
 
-  public canUpdateSomeProjects = (): boolean => {
-    // TODO: loop through all projects and check if the user has permission to update at least one
-    return this.checkProjectFilterPermission(
-      { projects: [] },
-      "manageProjects"
+  // Used to determine if we should show the Settings > Projects link in SideNav
+  public canManageSomeProjects = (): boolean => {
+    const projects: string[] = [""];
+
+    Object.keys(this.userPermissions.projects).forEach((key) =>
+      projects.push(key)
+    );
+
+    return projects.some((project) =>
+      this.checkProjectFilterPermission(
+        { projects: [project] },
+        "manageProjects"
+      )
     );
```

Comment:
In this case, we can't go with the simplified route exactly as suggested.

For this permission check, we want to know if the user has permission to manage projects either globally, or in at least 1 project. 

And in the case that the user has the permission globally, and they don't have any project-specific permissions, this will return false incorrectly, as the array built from `Object.keys(this.userPermissions.projects)` will be an empty array (since they don't have any project-level roles) - and calling `.some()` on an empty array always returns false.

In the original code, we're getting around this by seeding the first index of the array with an empty string. So even if the user doesn't have any project-level permissions, we call this.checkProjectFilterPermission at least once, and pass in an empty string for the project id, which then results in us checking the user's global role.

So, something like the following would work:
```ts
  public canManageSomeProjects = (): boolean => {
    if (
      Object.keys(this.userPermissions.projects).some((project) =>
        this.checkProjectFilterPermission(
          { projects: [project || ""] },
          "manageProjects"
        )
      )
    ) {
      return true;
    } else {
      return this.checkProjectFilterPermission(
        { projects: [] },
        "manageProjects"
      );
    }
  };
```
As this will check the user's project-level roles (if any) and if they don't have permission via a project-role, we'll check their global role.

This check is a bit of an oddball. We have the concept of a `readonly` type permission which handles this gracefully (we consider the user to have the permission if they have the permission either globally or in at least 1 project), but in this case, we don't want `manageProjects` to be a readonly type, we just want to check it as if it were a readonly type for the SideNav. So that if they have the permission globally or in at least 1 project, we show the `Projects` SideNav link.

Reply:
I do realize, especially after reading this, that there is some hidden logic here, that isn't super easy to identify.

We could do something like:
```ts
public canManageSomeProjects = (): boolean => {
// check user's global role first
if (this.checkProjectFilterPermission(
   { projects: [] },
   "manageProjects"
  ) {
  return true
 }
 
 // otherwise, check the user's project roles, if any
 return Object.keys(this.userPermissions.projects).some((project) =>
        this.checkProjectFilterPermission(
          { projects: [project] },
          "manageProjects"
        )
     );
 }
 ```